### PR TITLE
Remove and prevent trailing whitespace

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
- Remove various trailing whitespace.
- Add test that prevents trailing whitespace to be added.

One reason why this is important is https://github.com/ansible-collections/community.general/pull/10146#discussion_r2116335071. (According to https://github.com/ansible/ansible/issues/85237#issuecomment-2937597893 this is not a bug but a missing feature :shrug:.)

##### ISSUE TYPE
- Docs Pull Request
- Test Pull Request

##### COMPONENT NAME
various

### CODE REVIEW

File: tests/integration/targets/homebrew/tasks/casks.yml [hunk 1]
```diff
@@ -97,10 +97,10 @@
...
-  - name: Install cask using homelab/cask syntax 
-    homebrew: 
+  - name: Install cask using homelab/cask syntax
+    homebrew:
       package: "homebrew/cask/{{ package_name }}"
       state: present
       update_homebrew: false
-    become: true 
+    become: true
     become_user: "{{ brew_stat.stat.pw_name }}"
```
Comment (russoz) [inline tests/integration/targets/homebrew/tasks/casks.yml:106]:
```suggestion
    become_user: "{{ brew_stat.stat.pw_name }}"

```

File: tests/integration/targets/systemd_info/tasks/main.yml [hunk 2]
```diff
@@ -18,9 +18,9 @@
...
-    (ansible_distribution in ['RedHat', 'CentOS', 'ScientificLinux'] and ansible_distribution_major_version is version('7', '>=')) or 
-    ansible_distribution == 'Fedora' or 
-    (ansible_distribution == 'Ubuntu' and ansible_distribution_version is version('15.04', '>=')) or 
-    (ansible_distribution == 'Debian' and ansible_distribution_version is version('8', '>=')) or 
+    (ansible_distribution in ['RedHat', 'CentOS', 'ScientificLinux'] and ansible_distribution_major_version is version('7', '>=')) or
+    ansible_distribution == 'Fedora' or
+    (ansible_distribution == 'Ubuntu' and ansible_distribution_version is version('15.04', '>=')) or
+    (ansible_distribution == 'Debian' and ansible_distribution_version is version('8', '>=')) or
     ansible_os_family == 'Suse' or
     ansible_distribution == 'Archlinux'
```
Comment (russoz) [inline tests/integration/targets/systemd_info/tasks/main.yml:26]:
```suggestion
    ansible_distribution == 'Archlinux'

```

**Review by russoz — APPROVED** [review]
Looking good - just marked some end of files missing the newline.

**Comment by felixfontein** [comment]
@russoz thanks for reviewing!
